tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+
+import 'environment.dart';
+
+/// A function that returns true or false when given a [BuildConfig] and its
+/// name.
+typedef ConfigFilter = bool Function(String name, BuildConfig config);
```

Comment:
Personally (here and below) I'd just inline the function definition.

That avoids having to explain these generically "returns true or false" and you can just document how it's used in the (single?) function below.